databuilder/backends/base.py

```diff
@@ -36,7 +36,8 @@ def validate_contracts(cls):
         """
         for name, table in cls.tables.items():
             contract = table.implements
-            contract.validate_implementation(cls, name)
+            if contract is not None:
```

Comment:
Interesting. I'd say it's quite idiomatic to have `is None` checks rather than truthiness for variables with the type `Optional[T]`. In fact, relying on implicit boolean conversion has some [amusing edge cases](https://lwn.net/Articles/590299/).

Reply:
That's very funny. As I say, I don't like the auto-booleanizing for just this reason, so I'm happy to adjust my idioms.